compiler/parser/Parser.y

```diff
@@ -1373,12 +1374,17 @@ interface_body_decls
 interface_body_decl :: { Located InterfaceBodyDecl }
 interface_body_decl
   : ensure_decl                                  { sL1 $1 $ InterfaceEnsureDecl $1 }
+  | interface_view_type_decl { $1 }
   | 'choice' tycon OF_TYPE btype_ maybe_docprev arecord_with_opt interface_choice_body
         { sL (comb3 $2 $6 $>) $ InterfaceChoiceDecl (InterfaceChoiceSignature Nothing $2 $4 $6 $5) (unLoc $7) }
   | consuming_ 'choice' tycon OF_TYPE btype_ maybe_docprev arecord_with_opt interface_choice_body
         { sL (comb3 $3 $7 $>) $ InterfaceChoiceDecl (InterfaceChoiceSignature (Just (unLoc $1)) $3 $5 $7 $6) (unLoc $8) }
   | var OF_TYPE sigtype maybe_docprev { sL1 $1 $ InterfaceFunctionSignatureDecl $1 $3 $4 }
 
+interface_view_type_decl :: { Located InterfaceBodyDecl }
+interface_view_type_decl
+  : 'viewtype' ktypedoc { sLL $1 $> (InterfaceViewDecl $2) }
```

Comment:
I'm only bringing this up since I wasn't familiar with `ktypedoc`. The closest existing part of the template/interface syntax where we parse a type defined elsewhere would be choice return types, where we use `btype_`. The distinction seems to be that: 
* `ktype` is a possibly kind annotated `ctype`,
* `ctype` is a possibly constrained, possibly forall-quantified `type`
* `type` is a `btype` or `btype '->' ctype`

On the other side, `btype_` is "like 'btype' but excludes @arecord_with@".

Since I don't think we want kind-annotated (the kind would always be `Type`), constrained/quantified (type variables don't make sense here), function viewtypes (how??), nor `with` records, I think this would make more sense as `btype_`.

`btype_` doesn't have a `btype_doc` equivalent, but conceptually `ktypedoc` takes a comment for _the type_ whereas we want a comment for _the viewtype declaration_, so when we do add documentation support for viewtypes, the approach with `maybe_docprev` used for choice return types should work here too.

```suggestion
  : 'viewtype' btype { sLL $1 $> (InterfaceViewDecl $2) }
```